test/amqp/high_level_test.rb

```diff
@@ -134,4 +134,450 @@ def test_it_can_resubscribe_on_reconnect
       client.stop
     end
   end
+
+  def test_fanout_convenience_method
+    msgs = Queue.new
+    client = AMQP::Client.new("amqp://#{TEST_AMQP_HOST}").start
+    begin
+      e = client.fanout("test.fanout", auto_delete: true)
+      q = client.queue("test.fanout.queue")
+      q.bind("test.fanout", "")
+      q.subscribe do |msg|
+        msgs.push msg
+      end
+      e.publish("fanout message", "")
+
+      msg = msgs.pop
+      assert_equal "fanout message", msg.body
+      assert_equal "test.fanout", msg.exchange_name
+
+      e.delete
+      q.delete
```

Comment:
Looks like there's a risk of the queue leaking if the assertions fail, no?

(Same for all)